### ISSUE TITLE
docs: document meaning of generics

### ISSUE THREAD

**Issue by AlexElin** [angular/components#22121]
#### Documentation Feedback

In docs it would be helpful to have documentaion for declared TS generics.
For example, I open `MatDatepickerInputEvent` [docs](https://material.angular.io/components/datepicker/api#MatDatepickerInputEvent) and see `S` and `D` generics. 
And from the docs it's unclear what these `S` and `D` represent!

I think it should be documented similar to as Java does it.
For example, a [docs](https://docs.oracle.com/en/java/javase/15/docs/api/java.base/java/util/function/Function.html) of Java's standard type `Function<T,​R>` contain a section

> Type Parameters:
> T - the type of the input to the function
> R - the type of the result of the function

**Comment by jelbourn** [comment]
Yeah, we should do this. I've had it in our team backlog for a while, but it hasn't been a super high priority vs other work.

**Comment by jelbourn** [comment]
Reclassifying this as a bug instead of a feature

**Comment by mynameiszain** [comment]
Happy to take this on if it’s still available. Before I start, can you confirm whether the expected fix is in the API docs generator (dev-infra) or in the source API comments?
